src/repr/src/adt/datetime.rs

```diff
@@ -1493,8 +1502,12 @@ fn fill_pdt_from_tokens<'a, E: IntoIterator<Item = &'a TimeStrToken>>(
                     )
                 }
                 None => {
+                    // create signed copy of *val
+                    let sval = i64::try_from(i128::from(*val) * i128::from(sign))
+                        .map_err(|_| format!("Unable to parse value {} as a number: number too large to fit in target type", *val))?;
```

Comment:
You could try if this works:
```suggestion
                        .map_err(|_| format!("Unable to parse value {val} as a number: number too large to fit in target type"))?;
```

(I'm not sure if the dereference is needed for printing.)